### PR TITLE
Fixed PHP 7 failures

### PR DESCRIPTION
**Pull request by XWB**

### CODE REVIEW

File: Tests/Helper/NumberHelperTest.php [hunk 1]
```diff
@@ -97,6 +97,12 @@ public function testArguments()
      */
     public function testExceptionOnInvalidParams()
     {
+        if (PHP_VERSION_ID >= 70000) {
```
Comment (Soullivaneuh) [inline Tests/Helper/NumberHelperTest.php]: Please use `version_compare`.

File: Tests/Helper/NumberHelperTest.php [hunk 2]
```diff
@@ -97,6 +97,12 @@ public function testArguments()
...
+            $this->expectedException('\IntlException');
+        } else {
+            $this->expectedException('\RuntimeExceptio');
```
Comment (Soullivaneuh) [inline Tests/Helper/NumberHelperTest.php]: Missing `n`.

File: Tests/Helper/NumberHelperTest.php (hunk 1 above)
Comment (Koc) [inline Tests/Helper/NumberHelperTest.php]: Why?

Comment (OskarStark) [inline Tests/Helper/NumberHelperTest.php]: IMO the Symfony guys use the PVP_VERSION_ID to compare against a PHP version

Comment (XWB) [inline Tests/Helper/NumberHelperTest.php]: PHP_VERSION_ID should be fine, it's how Symfony does it.

**Comment by OskarStark** [comment]
can you please rebase this branch @XWB

**Comment by XWB** [comment]
@OskarStark Done.

**Comment by rande** [comment]
@OskarStark Can you find an elegant way to throw a RuntimeException ? so the code is BC.

**Comment by XWB** [comment]
How about splitting `testExceptionOnInvalidParams`?

**Comment by rande** [comment]
@XWB inside the lib  not the test. IntlException inherit from Exception and not RuntimeException so we need to make sure the SonataIntlBundle keep throw RuntimeException to avoid BC break with existing code.

File: Tests/Helper/NumberHelperTest.php (hunk 1 above)
Comment (Soullivaneuh) [inline Tests/Helper/NumberHelperTest.php]: So I don't understand why they are using this.

IMO, [`version_compare`](http://php.net/manual/fr/function.version-compare.php) is much easier to read and understand.

cc @rande what is the convention in sonata for that?

**Comment by OskarStark** [comment]
any news @XWB

**Comment by OskarStark** [comment]
ping @XWB

**Comment by OskarStark** [comment]
btw i think there is no possibility to stay BC and would introduce this in the next major

what do you think @rande, if yes @XWB please add a UPGRADE notice

**Comment by rande** [comment]
It is possible, we just need to catch the IntlException and throw a new RuntimeException.

**Comment by XWB** [comment]
PR updated.

File: Tests/Helper/NumberHelperTest.php [hunk 3]
```diff
@@ -98,7 +98,16 @@ public function testArguments()
     public function testExceptionOnInvalidParams()
     {
         // https://wiki.php.net/rfc/internal_constructor_behaviour
-        $formatter = new \NumberFormatter('FR', -1);
+        if (PHP_VERSION_ID >= 70000) {
```
Comment (core23) [inline Tests/Helper/NumberHelperTest.php]: IMO you could remove the version check here

**Comment by XWB** [comment]
@core23 Done.

**Comment by core23** [comment]
Please check travis @XWB

**Comment by XWB** [comment]
@core23 Not sure how travis error is related to this PR?

**Comment by core23** [comment]
You're right. The problems is the **SonataUserBundle**. Created a new PR for this https://github.com/sonata-project/SonataIntlBundle/pull/109

**Comment by core23** [comment]
Thank you  @XWB

**Pull request merged**